rclpy/rclpy/wait_for_message.py

```diff
@@ -22,6 +25,8 @@ def wait_for_message(
     msg_type,
     node: 'Node',
     topic: str,
+    *,
+    qos_profile: Union[QoSProfile, int] = 1,
     time_to_wait=-1
```

Comment:
we had a quick chat about this, and this actually breaks the downstream application. (@cottsay thanks!)
If any downstream code previously called `wait_for_message` using four positional arguments, expecting the fourth to be `qos_profile`, those calls would no longer work as intended, leading to exception.

i think what we can do is, adding `qos_profile` as a positional or optional argument after `time_to_wait`.

```suggestion
    time_to_wait=-1,
    qos_profile: Union[QoSProfile, int] = 1
```

This way, the change is backward-compatible, and `qos_profile` can still be passed as a positional argument if needed.

So far we do not have any request to backport this Iron, so we will close this PR.